proposal/nested-map-functions.md

```diff
@@ -0,0 +1,391 @@
+# Nested Map Functions: Draft 1.0
+
+*[(Issue)](https://github.com/sass/sass/issues/1739)*
+
+This proposal updates the built-in `sass:map` module to better support merging,
+setting, and getting elements from nested maps.
+
+## Table of Contents
+
+* [Background](#background)
+* [Summary](#summary)
+* [Functions](#functions)
+  * [`get()`](#get)
+  * [`has-key()`](#has-key)
+  * [`set()`](#set)
+  * [`merge()`](#merge)
+  * [`deep-merge()`](#deep-merge)
+  * [`deep-remove()`](#deep-remove)
+
+## Background
+
+> This section is non-normative.
+
+Variables have always been a key feature of the Sass language. But these days,
+design systems and component libraries form the basis of most CSS projects --
+with well organized _design tokens_ as the foundation. While Individual token
+variables can be quite useful, the ability to group tokens into strucutred and
+meaningful relationships is essential for creating resilient systems.
+
+There are many ways to group tokens. The popular Style Dictionary recommends
```

Comment:
Might be worth linking https://amzn.github.io/style-dictionary/#/?id=style-dictionary